datafusion/core/src/physical_plan/planner.rs

```diff
@@ -964,7 +965,27 @@ pub fn create_physical_expr(
                 input_schema,
                 execution_props,
             )?;
-            binary(lhs, *op, rhs, input_schema)
+            match (
```

Comment:
I like this approach to use a specialized operator

I think over the long term, the ideal outcome would be to extend the arrow temporal kernels https://github.com/apache/arrow-rs/blob/master/arrow/src/compute/kernels/temporal.rs with the actual calculation logic.

However, incubating the kernels within datafusion seems like a great idea